website_sale_qty/static/src/js/website_sale_qty.js

```diff
@@ -0,0 +1,41 @@
+/* Copyright 2016-2017 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html) */
+
+odoo.define('website_sale_qty', function(require) {
+    'use strict';
+
+    var base = require('web_editor.base');
+    base.ready().done(function() {
+        var radios = $('.js-pqt-input');
+
+        if (radios.length > 0) {
+            var radioMap = {};
+            _.each(radios, function(element) {
+                var $element = $(element);
+                radioMap[$element.val()] = $element;
+            });
+            var quantityInput = $('.css_quantity input');
+            var quantityToggles = $('.css_quantity a');
+
+            radios.click(function(event) {
+                var newValue = $(event.currentTarget).val();
+                quantityInput.val(newValue).trigger('change');
+            });
+
+            var updateRadios = function() {
+                radios.prop('checked', false);
+
+                var newValue = quantityInput.val();
+                if (newValue in radioMap) {
+                    radioMap[newValue].prop('checked', true);
+                }
+            };
+
+            quantityInput.on('input', updateRadios);
+            quantityToggles.click(function() {
+                // This allows website_sale.website_sale to update the input first
```

Comment:
My understanding is that there are no race conditions in JS. If the same event leads to multiple messages being added to the queue using ``setTimeout`` with a delay of ``0``, then their order of execution will be unclear, but it'll still be sequential. And since ``setTimeout`` always adds messages to the queue, it feels like overkill to explain this in a comment.